gbpservice/nfp/configurator/agents/agent_base.py

```diff
@@ -81,13 +81,23 @@ def process_request(self, sa_req_list, notification_data):
                                    data=args_dict, key=None)
             self.sc.post_event(ev)
         else:
-            sa_req_list[0]['context'].update(
-                {'notification_data': notification_data})
-            sa_req_list[0]['context'].update(
-                {'resource': sa_req_list[0]['resource']})
-            getattr(self, sa_req_list[0]['method'])(
-                sa_req_list[0]['context'],
-                **sa_req_list[0]['kwargs'])
+            agent_info = sa_req_list[0]['agent_info']
+            # Renaming the neutron context in resource data of *aaS to context.
+            # Adding agent_info which contains information required for
+            # demux and response data in agent to neutron_context in *aaS
+            if not sa_req_list[0]['is_generic_config'] and not (
```

Comment:
Generic config requests will not be heat/ansible right? Why do we need to explicitly check for this?